lib/orogen/gen/project.rb

```diff
@@ -814,6 +814,7 @@ def tasklib_used_task_libraries
                     result = self_tasks.inject(Set.new) do |set, task|
                         set | task.used_task_libraries
                     end
+                    result.merge(used_task_libraries)
                     result.to_a.sort_by(&:name)
```

Comment:
`self#used_task_libraries` is a superset of the `result` created from `self_tasks`, so you can simply replace the whole method with `used_task_libraries.to_a.sort_by(&:name)`